src/components/PoolTab.js

```diff
@@ -1,32 +1,47 @@
-import React, { useEffect, useState } from "react";
+import React, { useEffect, useState, useContext } from "react";
 
-import { convertToE24Base, browsePools, poolInfo } from "../services/near-nep21-util";
+import { convertToE24Base, browsePools, poolInfo, sharesBalance } from "../services/near-nep21-util";
 
-import PoolInputCards from "./PoolInputCards"
 import PoolInfoCard from "./PoolInfoCard"
+import AddLiquidityModal from "./AddLiquidityModal"
+
+import { TokenListContext } from "../contexts/TokenListContext";
 
 import Button from 'react-bootstrap/Button';
 
 import { BsPlus } from "react-icons/bs";
 
-import styled from "@emotion/styled";
-const Hr = styled("hr")`
-  border-top: 1px solid ${props => props.theme.hr}
-`;
-
 export default function PoolTab() {
 
   const [pools, setPools] = useState([]);
 
+  // Token list state
+  const tokenListState = useContext(TokenListContext);
+
   async function fetchPools() {
     browsePools()
```

Comment:
Let's use `await` - the function is already marked async, instead of callbacks, and don't use `map` if you put local function longer than 5 lines (use for loop instead):
```
async function fetchPools() {
  const pools = await browsePools();
  for (let p of pools) {
    const pi = await poolInfo(p);
    ...
  }
}
```